com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -8,13 +8,28 @@
 
 namespace Unity.MLAgents.Inference
 {
+
+    internal enum CheckType
+    {
+        Info = 0,
+        Warning = 1,
+        Error = 2
+    }
+
+    internal class FailedCheck
+    {
+        public CheckType CheckType;
+        public string Message;
+    }
+
     /// <summary>
     /// Prepares the Tensors for the Learning Brain and exposes a list of failed checks if Model
     /// and BrainParameters are incompatible.
     /// </summary>
     internal class BarracudaModelParamLoader
     {
-        const long k_ApiVersion = 2;
+        public const long k_ApiVersion = 3;
+        public const long k_ApiVersionLegacy = 2;
```

Comment:
I need help with this. For now version 2 is the only legacy, but that won't be the case for ever. I do not want to make it a list because not all legacy versions will be treated the same.
We could enforce that there is always only one current version and one legacy version (corresponding to the previous major release).
Thoughts ?

Reply:
You could do something like
```csharp
enum ModelApiVersion {
  MLAgents1_0 = 2,
  MLAgents2_0 = 3,
  MinSupportedVersion = MLAgents1_0
}
```
Then raise errors if `version < (int) MinSupportedVersion`, and switch on the legacy discrete actions based on `MLAgents1_0`.